internal/storage/mock/testify_mock_bucket.go

```diff
@@ -124,3 +125,11 @@ func (m *TestifyMockBucket) CreateFolder(ctx context.Context, folderName string)
 	}
 	return nil, args.Error(1)
 }
+
+func (m *TestifyMockBucket) SyncObject(ctx context.Context, fileName string, srcObject *gcs.Object, content gcsx.TempFile) (o *gcs.Object, err error) {
```

Comment:
also fakes are preferred over mocks. Why do you want to have a mock for syncObject when the methods used by internally has fake implementation.

Reply:
In file_test.go we are creating the content (temp file) before hand for the tests. This won't work for buffered writes as I want to ensure that in case of failure temp file is created. That is why I needed mocks. Do you think there is a better way to do this?